### DISCUSSION TITLE
Possible voting changes to quarterly planning

### DISCUSSION THREAD

**Discussion by wbamberg** [openwebdocs/project#72] [General]
In the OWD/MDN content planning meeting today we had an interesting discussion about possible changes to the way we decide what OWD staff should work on in the next quarter.

Here I'll write up and contextualize the discussion, and see if we will have consensus for making any changes.

## Current process

The current process is something like this:
* anyone can propose projects for OWD by filing issues in this repo
* proposers are "encouraged" to score their projects using our [prioritization criteria](https://github.com/openwebdocs/project/blob/main/steering-committee/prioritization-criteria.md)
* proposals are pre-screened
* before the end of a quarter we have (at least) one meeting including OWD staff and the steering committee, in which we talk through the proposals
* finally we vote: each SC representative gets a vote (note that each SC organization has two representatives) and OWD staff reach get a vote
* votes are tallied and projects ranked accordingly, and OWD staff pick the next quarter's projects accordingly.

## Problems

A couple of problems were raised:

* (Estelle) if people only feel strongly about certain projects, and are essentially neutral about others, they might not feel able to vote for those projects. But then these projects score zero points.
* (Daniel) the voting system seems gameable and encourages people to think about what either people are going to vote for. (note though that nobody, including OWD staff, has any prior knowledge of how anyone else votes)
* (Reeza) do we have concerns that if projects stay too long in the backlog they should be bumped up or removed? How do we avoid an ever-growing list of potential projects?

## Some possible solutions

* rather than everyone ranking all projects, could each person get a bunch of votes and be able to assign them to projects as they wish (so for example if there were 15 projects, each person might get 15 votes but could choose to split all their votes across their three favourite projects.

* consider splitting projects into big projects (which get voted on) and little projects (which can get taken on at the discretion of OWD staff. This is intended to help the fact that little projects could get squeezed out by bigger projects.

## Raw notes

Estelle: what if people don’t have opinions about some projects? If they don’t vote for everything,projects they don’t vote for score zero.

Ruth: should we have weighted votes for the things people really want. Reeza: what if people only get to choose three (say) projects?

Dominic: also some projects are small, and we might want to do these too even if they are not so high priority. Perhaps we want to vote separately on big and small projects.

Daniel: the voting system seems gameable and encourages people to think about what either people are going to vote for. Daniel: also expect some flexibility in interpreting results: we might do 5 and 6 if they are cheap, at the expense of 4 which is expensive.

Florian: the main thing is the signal we get from the SC about what they think is important. 

Lola: OWD are in the best position to vote, as they have more insight than other SC members. Could OWD vote first and present to SC for feedback? I.e. switch to seeking-consensus.

Dominic: liked having SC provide input first, and got good feedback from inside FB, and this would be harder if he already had a curated list. Appreciated having visibility of votes, expect that this might enable horse-trading in future quarters.

Florian: the “elevator pitch” meeting was great. Estelle: making every proposal have a champion helps people get a sense of what's best for OWD.

Reeza: do we concerns that if things stay too long in the backlog they should be bumped up or removed? Estelle: if each proposal has a champion can they own the problem of dealing with a project that’s not been prioritized.

Will: should we help with the problem that people have to vote for everything? Daniel: give people 16 votes and let them distribute them? Estelle: but can this make it easier for some people to win projects by using all their votes. Dominic: likes points, but thinks it will penalise small projects. Maybe they should be put in a separate stream and  left more to the whims of OWD staff members?

Florian: if anyone has contacts with orgs who do this sort of thing we are interested in more/better ideas. Will: will summarize the discussion for the next SC meeting.

**Comment by foolip** [comment]
> the voting system seems gameable and encourages people to think about what either people are going to vote for.

@ddbeck can you elaborate on this? Given that we each get to rank as many options as we like, would tactical voting actually work? At least for my part, I didn't think about what other would vote for, because I assumed some off-the-shelf voting system that doesn't have this problem is in use.

P.S. @dontcallmedom cautioned us in the GC call not to get too deep into voting systems, as it can be a time sink, I hope this question doesn't open that door too widely.

**Reply by estelle** [reply]
My suggestion: we should give every eligible voter a set number votes or points to distribute to the proposals they support, putting a very low max number of points per any one proposal. Maybe a max of 20% of their votes/points to any one topic. Then the priority becomes the proposals with the largest number of points

**Reply by dominiccooney** [reply]
> can you elaborate on this?

This is paraphrasing, but @ddbeck mentioned he was playing around with the spreadsheet after the vote, and noticed a single voter could cause big shifts in the ranks of "middling" proposals. That's where the cut-off for which proposals get done are.

> would tactical voting actually work?

Individual tactical voting is hard but I think it would be very effective in groups. I don't view this as necessarily negative since it gets people discussing and maybe strengthening proposals.

> I didn't think about what other would vote for, because I assumed some off-the-shelf voting system that doesn't have this problem is in use.

I don't think this is a safe assumption. I also think we casually toss around the word "problem" but people have different views on what's strategic versus manipulative. I liked how @ddbeck approached this as (again paraphrasing) he didn't think one set of votes should be able to have such an apparently outsized influence.

+1 to @dontcallmedom 's observation that voting systems are a rabbit hole. I am fine with any of these: Keep the status quo; try @estelle 's proposal, which removes the burden of ranking the tail of projects and gives more signal about relative importance; try rank choice voting.

**Reply by foolip** [reply]
@Elchi3 can you explain what the current voting system is, how you go from individual rankings to an overall ranking? Like @dominiccooney I'm probably happy to keep it or try something else, but knowing what we have now would be a good start.

**Reply by Elchi3** [reply]
I wasn't doing anything fancy, here's the spreadsheet: https://docs.google.com/spreadsheets/d/1zuaBQ__4_2-s2LiiK9A4-XaQUIR0ucN7O1OYWQ1TsfE/edit#gid=0. I looked at average and median and then sorted it. I'm happy to do some better calc.

**Reply by foolip** [reply]
I see, just averaging the ranks numerically. That probably gets close to a more formal method, but maybe trying an implementation of https://en.wikipedia.org/wiki/Schulze_method would be worthwhile. If that gives a different ranking than the spreadsheet, then I'd assume the Schulze method gives the better results.

**Reply by Rumyra** [reply]
I had a play around with this - yes out of actual curiosity 🥺 - found an npm module for schulze method:

https://codepen.io/Rumyra/pen/vYJMMxm?editors=1001

Results are outputted to the console and I _believe_ they've changed. I don't recall the 'use webref' proposal being quite as high.

I also think if we were to use this, creating a 'weighted' voting system (e.g. I'm giving 3 proposals a weight of 1/high, 4 a weight of 2/mid and the rest 3/low) would work best - but I'm assuming that based on reading about the Schulze method tho 👍

**Reply by foolip** [reply]
> I also think if we were to use this, creating a 'weighted' voting system (e.g. I'm giving 3 proposals a weight of 1/high, 4 a weight of 2/mid and the rest 3/low) would work best - but I'm assuming that based on reading about the Schulze method tho 👍 

I'm not too sure, but I think this isn't what the numbers mean. Rather, you can rank as many options as you like and use the same number for options you're undecided between.

This is still different from our approach where a complete ordering is required, however.
